### PR TITLE
Allow searching users by their email instead of doing it only by nickname

### PR DESCRIPTION
**Pull request by pypingou**

### CODE REVIEW

File: fas/user.py [hunk 1]
```diff
@@ -600,14 +600,17 @@ def dump(self, search=u'a*', groups=''):
...
+    def list(self, search=u'a*', fields=None, limit=None, status=None,
+            by_email=None):
         '''List users
 
         :kwarg search: Limit the users returned by the search string.  * is a
             wildcard character.
         :kwarg fields: Fields to return in the json request.  Default is
             to return everything.
         :kwargs status: if specified, only returns accounts with this status.
+        :kwargs status: if true or 1, the search is done by email instead of
```
Comment (ralphbean) [inline fas/user.py]: typo here.  should be ``by_email`` not ``status``.

**Comment by ralphbean** [comment]
How about adding ``by_ircnick`` too?  :)

File: fas/user.py (hunk 1 above)
Comment (pypingou) [inline fas/user.py]: Good catch

**Comment by pypingou** [comment]
Sounds like a good idea we should just remember that for now FAS2 doesn't enforce unique irc nick :)

**Comment by pypingou** [comment]
One thing I was thinking of for this PR, check if there is a '%' in ``re_search`` and only use ``ilike()`` if there is, other use a plain ``==``

**Comment by pypingou** [comment]
``by_ircnick`` implemented in stg at https://admin.stg.fedoraproject.org/accounts/user/list/kev*?by_ircnick=1

**Comment by ralphbean** [comment]
LGTM!  :+1:

**Pull request merged**